packages/design-system/src/components/Button/variations/ButtonTertiary.tsx

```diff
@@ -1,12 +1,25 @@
 import React, { forwardRef, Ref } from 'react';
-import ButtonPrimitive, { BaseButtonProps } from '../Primitive/ButtonPrimitive';
+import ButtonPrimitive, { AvailableSizes, BaseButtonProps } from '../Primitive/ButtonPrimitive';
 
 import styles from './ButtonTertiary.module.scss';
+import { ButtonDestructivePropsType } from './ButtonDestructive';
 
-export type ButtonTertiaryPropsType = Omit<BaseButtonProps, 'className'>;
+export type ButtonTertiaryPropsType<S extends AvailableSizes> = Omit<
+	BaseButtonProps<S>,
+	'className' | 'size'
+> & {
+	size?: S;
+};
 
-const ButtonTertiary = forwardRef((props: ButtonTertiaryPropsType, ref: Ref<HTMLButtonElement>) => {
+function Tertiary<S extends AvailableSizes>(
+	props: ButtonTertiaryPropsType<S>,
+	ref: Ref<HTMLButtonElement>,
+) {
```

Comment:
Don't we need to 
`const { size = 'M', ...rest } = props;` on this button too ?

Reply:
TS didn't tell me to 🤔 I might remove the destruct on the other ones if they are overkill. Thanks for spotting it.

Reply:
Yep, not needed, thenks